app/shared/models/build.rb

```diff
@@ -73,5 +73,10 @@ def status=(new_value)
       raise "Invalid build status '#{new_value}'" unless BUILD_STATUSES.include?(new_value)
       @status = new_value
     end
+
+    def link_to_remote_commit
+      # TODO: The line below doesn't work for `git://` URLs (which we currently don't support anyway)
+      "#{project.repo_config.git_url}/commit/#{sha}"
```

Comment:
Right now this is very "hard coded", took from the summary view.
Eventually, I would say that we should use the provider+repo_config in order to build this url, for example: GitHub uses "/commit/xx", BitBucket uses "/commits/xx", other provider would probably have a different URL too. Also, at the moment, it looks like that repo_config is strictly linked to `GitRepoConfig` (which despite the name, it seems to be for GitHub provider only, as specified by `provider_credential_type_needed`. Anyway, I only started to look into the project very recently so my understanding could be wrong :) Thoughts?

Reply:
I'd try to make a switch case depending on the current provider credential type, raise an exception when other provider credential is not github for now, but that way we avoid getting that hard-coded

Reply:
@adellibovi yeah, good find: `GitRepoConfig` is tied to `GitHub`, it probably should be renamed

Reply:
@taquitos @minuscorp, What if `RepoConfig` will implement a method to return the right "link"? Of course, assuming `GitRepoConfig` is for `GitHub`. The idea is to avoid having to implement different cases of switches and instead implementing a method to the RepoConfig which is already "provider"-related. Wouldn't it be better? Also, since we have the "fullName" we may actually create the link without relying on the `git_url` and therefor fixes the `TODO` too.

Reply:
Yeah, that makes sense. When we tie a Class to a provider, we can then just create a method that returns the correct link without even using a switch statement. Good idea 👍